### PR TITLE
Scoutsuite docs

### LINKED ISSUE #1010: Add ScoutSuite documentation.
## Documentation request

Add documentation about ScoutSuite in our documentation hub.

### Add a new page 🆕
Add the bulk of documentation to a new page in `Reference > ScoutSuite`. Link to this information should appear in `Usage > Use Cases > Zero Trust assessment`

**What documentation you'd like us to add?**
Add a link to ScoutSuite project, elaborate on how to setup and run this assessment.

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Fixes #1010 
Improves UI of ScoutSuite rule display a bit.
Cleans up tutorial screenshots for key setup a bit.

image

### CODE REVIEW

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: "Scoutsuite"
+date: 2021-03-02T16:23:06+02:00
+draft: false
+---
+
+### What is ScoutSuite?
+
+<a href="https://github.com/nccgroup/ScoutSuite" target="_blank" >Scout Suite</a> is an open-source cloud security-auditing tool.
+It queries the cloud API to gather configuration data of the cloud infrastructure. Based on configuration
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:10]:
```suggestion
It queries the cloud API to gather configuration data from the cloud infrastructure. Based on configuration
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+data gathered ScoutSuite shows security issues and risks present in your infrastructure.
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:11]:
```suggestion
data gathered, ScoutSuite shows security issues and risks present in your infrastructure.
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+### Which cloud providers are supported?
+
+So far the Infection Monkey only supports AWS.
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:15]:
```suggestion
Today, the Infection Monkey only supports AWS.
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+### How to enable ScoutSuite?
+
+First of all, Infection Monkey needs access to your cloud API. You can provide access
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:19]:
```suggestion
First, Infection Monkey needs access to your cloud API. You can provide access
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+in the following ways:
+
+ - Provide access keys:
+    - Create a new user with ReadOnlyAccess and SecurityAudit policies and generate keys
+    - Generate keys for your current user (faster but less secure)
+ - Configure AWS CLI:
+    - If the command-line interface is available on the Island, it will be used to access 
+    the cloud API
+    
+More details about configuring ScoutSuite can be found in the tool itself, by choosing
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:29]: Maybe add a screenshot that shows the Run Monkey page as it appears today.

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+More details about configuring ScoutSuite can be found in the tool itself, by choosing 
+"Cloud Security Scan" in the "Run Monkey" options. 
+
+After you're done with the setup, make sure that a checkmark appears next to the AWS option to 
+verify that ScoutSuite can access the API.
+
+![Successfull setup indicator](/images/reference/zero-trust/scoutsuite_aws_configured.png 
+"Successful setup indicator")
+
+### How to run cloud scan?
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:38]:
```suggestion
### How to run cloud security scan
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    - Generate keys for your current user (faster but less secure)
+ - Configure AWS CLI:
+    - If the command-line interface is available on the Island, it will be used to access 
+    the cloud API
+    
+More details about configuring ScoutSuite can be found in the tool itself, by choosing 
+"Cloud Security Scan" in the "Run Monkey" options. 
+
+After you're done with the setup, make sure that a checkmark appears next to the AWS option to 
+verify that ScoutSuite can access the API.
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:33]:
```suggestion
After you're done with the setup, make sure that a checkmark appears next to the AWS option. This 
verifies that ScoutSuite can access the API.
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+<a href="https://github.com/nccgroup/ScoutSuite" target="_blank" >Scout Suite</a> is an open-source cloud security-auditing tool.
+It queries the cloud API to gather configuration data of the cloud infrastructure. Based on configuration
+data gathered ScoutSuite shows security issues and risks present in your infrastructure.
+
+### Which cloud providers are supported?
+
+So far the Infection Monkey only supports AWS.
+
+### How to enable ScoutSuite?
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:17]:
```suggestion
### How to enable ScoutSuite
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+### How to run cloud scan?
+
+If you have successfully configured cloud scan, once the Monkey Agent is run **on the Island**, 
+the cloud infrastructure will get scanned. To make this happen, you can simply click on "From Island" 
+in the run options. The scope of network scan and other activities you configured the Agent to 
+do are irrelevant for cloud security scan, except 
+**Monkey Configuration -> System info collectors -> AWS collector** which needs to remain **enabled**.
+
+### How to assess cloud scan results?
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:46]:
```suggestion
### How to assess cloud scan results
```

File: docs/content/reference/zero-trust/scoutsuite.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+![Successfull setup indicator](/images/reference/zero-trust/scoutsuite_aws_configured.png 
+"Successful setup indicator")
+
+### How to run cloud scan?
+
+If you have successfully configured cloud scan, once the Monkey Agent is run **on the Island**, 
+the cloud infrastructure will get scanned. To make this happen, you can simply click on "From Island" 
+in the run options. The scope of network scan and other activities you configured the Agent to 
+do are irrelevant for cloud security scan, except 
+**Monkey Configuration -> System info collectors -> AWS collector** which needs to remain **enabled**.
```
Comment (mssalvatore) [inline docs/content/reference/zero-trust/scoutsuite.md:44]:
```suggestion
If you have successfully configured the cloud scan, Infection Monkey will scan
your cloud infrastructure when the Monkey Agent is run **on the Island**. You
can simply click on "From Island" in the run options to start the scan. The
scope of the network scan and other activities you may have configured the Agent
to perform are ignored by the cloud security scan, except **Monkey
Configuration -> System info collectors -> AWS collector**, which needs to
remain **enabled**.
```

File: docs/content/reference/zero-trust/scoutsuite.md (hunk 1 above)
Comment (VakarisZ) [reply docs/content/reference/zero-trust/scoutsuite.md]: This sentence tries to say that the data gathered is about cloud infrastructure, not that it's gathered from cloud infrastructure. Fixed to `It queries the cloud API to gather configuration data about the cloud infrastructure.`

File: docs/content/reference/zero-trust/scoutsuite.md (hunk 3 above)
Comment (VakarisZ) [reply docs/content/reference/zero-trust/scoutsuite.md]: Fixed to `Currently`, as we refer to this point in time, not a particular time unit. Could be removed altogether though

File: docs/content/reference/zero-trust/scoutsuite.md (hunk 8 above)
Comment (VakarisZ) [reply docs/content/reference/zero-trust/scoutsuite.md]: Reworded all headers

**Review by VakarisZ — COMMENTED** [review]
Ready for re-review

File: docs/content/usage/integrations/scoutsuite.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+can simply click on "From Island" in the run options to start the scan. The
+scope of the network scan and other activities you may have configured the Agent
+to perform are ignored by the ScoutSuite integration, except **Monkey
+Configuration -> System info collectors -> AWS collector**, which needs to
+remain **enabled**.
+
+
+### Assessing scan results
+
+After the scan is done, ScoutSuite results will be sorted and applied to the ZeroTrust Extended framework
```
Comment (mssalvatore) [inline docs/content/usage/integrations/scoutsuite.md:56]:
```suggestion
After the scan is done, ScoutSuite results will be categorized according to the ZeroTrust Extended framework 
```

File: docs/content/usage/integrations/scoutsuite.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+Configuration -> System info collectors -> AWS collector**, which needs to
+remain **enabled**.
+
+
+### Assessing scan results
+
+After the scan is done, ScoutSuite results will be sorted and applied to the ZeroTrust Extended framework 
+and displayed as a part of the ZeroTrust report. The main difference between Infection Monkey findings and 
+ScoutSuite findings is that ScoutSuite findings contain security rules. To see which rules were 
+checked click on the "Rules" button next to the relevant test. You'll see a brief overview of the rules
```
Comment (mssalvatore) [inline docs/content/usage/integrations/scoutsuite.md:59]:
```suggestion
checked, click on the "Rules" button next to the relevant test. You'll see a brief overview of the rules 
```

File: docs/content/usage/integrations/scoutsuite.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+remain **enabled**.
+
+
+### Assessing scan results
+
+After the scan is done, ScoutSuite results will be sorted and applied to the ZeroTrust Extended framework 
+and displayed as a part of the ZeroTrust report. The main difference between Infection Monkey findings and 
+ScoutSuite findings is that ScoutSuite findings contain security rules. To see which rules were 
+checked click on the "Rules" button next to the relevant test. You'll see a brief overview of the rules 
+related to the test and a list of those rules. Expand a rule to see its description, remediation and
```
Comment (mssalvatore) [inline docs/content/usage/integrations/scoutsuite.md:60]: "You'll see a brief overview of the rules related to the test and a list of those rules."

^ This is a little unclear. Why do we see an overview and a list? Is it a list of the rules that failed? Is it a list of all rules?

Maybe we can just get rid of "and a list of those rules" it it's redundant.

File: docs/content/usage/integrations/scoutsuite.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+### Assessing scan results
+
+After the scan is done, ScoutSuite results will be sorted and applied to the ZeroTrust Extended framework 
+and displayed as a part of the ZeroTrust report. The main difference between Infection Monkey findings and 
+ScoutSuite findings is that ScoutSuite findings contain security rules. To see which rules were 
+checked click on the "Rules" button next to the relevant test. You'll see a brief overview of the rules 
+related to the test and a list of those rules. Expand a rule to see its description, remediation and 
+more details about resources flagged. Each flagged resource has a path so you can easily locate 
+them in the cloud and change the value that is deemed insecure.
```
Comment (mssalvatore) [inline docs/content/usage/integrations/scoutsuite.md:62]:
```suggestion
it in the cloud and remediate the issue.
```

File: docs/content/usage/integrations/scoutsuite.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+
+### About ScoutSuite
+
+<a href="https://github.com/nccgroup/ScoutSuite" target="_blank" >Scout Suite</a> is an open-source cloud security-auditing tool.
+It queries the cloud API to gather configuration data. Based on configuration
+data gathered, ScoutSuite shows security issues and risks present in your infrastructure.
+
+### Supported cloud providers
+
+Currently, the Infection Monkey only supports AWS.
```
Comment (mssalvatore) [inline docs/content/usage/integrations/scoutsuite.md:17]: In the context of scoutsuite, this makes sense, but I wonder if someone may still find it a bit confusing. What about,

"Currently, the Infection Monkey can only use ScoutSuite to assess AWS environments."

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Just a few changes.

File: docs/content/usage/integrations/scoutsuite.md (hunk 13 above)
Comment (VakarisZ) [reply docs/content/usage/integrations/scoutsuite.md:60]: Reworded

File: docs/content/usage/integrations/scoutsuite.md (hunk 15 above)
Comment (VakarisZ) [reply docs/content/usage/integrations/scoutsuite.md:17]: Reworded

**Pull request merged**